packages/core/src/components/Toggle/types.ts

```diff
@@ -26,4 +27,8 @@ export interface ToggleProps extends InputProps {
     labelWeight?: FontWeights;
     /** Use if required in any form for example user needs to check 'I Agree' */
     required?: boolean;
+    /** Icon to show when toggle is in OFF state */
+    iconRight?: ReactElement;
+    /** Icon to show when toggle is in ON state */
+    iconLeft?: ReactElement;
```

Comment:
Can we use OnIcon and OffIcon to be more clear? Also this is the naming convention in the styled file, we can keep it consistent here

Reply:
The reasoning for this was, while `offIcon` would make sense, but `onIcon` will cause conflict with `onClick`, `onScroll` and all the other actions starting with `on-`

Reply:
can please add the screenshots of the toggle in the PR description.
@abhishekashyap

Reply:
Yes, I have added the screenshots. You can also view the chromatic storybooks deployment here: https://610a8c2fe7a1bd0039a44070-cvysngcsgn.chromatic.com/?path=/story/core--toggle-with-icon